WBooks/Screens/Library/LibraryViewController.swift

```diff
@@ -41,12 +40,39 @@ class LibraryViewController: BaseViewController {
         setLeftButtonImage(customImage: UIImage.notificationsIcon)
         setRightButtonImage(customImage: UIImage.searchIcon)
         setupTableView()
+        updateLoadingDisplay(loading: true)
+        _libraryViewModel.getBooks(onSuccess: fetchDataSuccess, onError: fetchDataError)
     }
     
     func setupTableView() {
         _view.booksTable.delegate = self
         _view.booksTable.dataSource = self
         _view.booksTable.register(cell: CustomBookCell.self)
+
+            
+        }
+    func fetchDataSuccess() {
+        _view.booksTable.reloadData()
+        updateLoadingDisplay(loading: false)
+    }
+    
+    func fetchDataError() {
+        self._view.booksTable.reloadData()
```

Comment:
unnecessary self. 
Actually you should show an error here. You have the same here as `fetchDataSuccess`

Reply:
😱